Upgrade TYPO3 v13

sebastian-altenburg-nr opened the pull request:
Upgrades the extension to TYPO3 v13.

Commit:
NEXT-44: change composer.json

CybotTM commented:
Hi @sebastian-altenburg-nr , please resolve the merge conflicts by rebasing the PR.

Thanks.

Commit:
NEXT-77: Prepare for rebase

CybotTM commented on Classes/Seo/ImagesXmlSitemapDataProvider.php:
```diff
@@ -37,57 +41,69 @@ class ImagesXmlSitemapDataProvider extends AbstractXmlSitemapDataProvider
             throw new MissingConfigurationException(
                 'No configuration found for sitemap ' . $this->getKey(),
-                1652249698
+                1_652_249_698
```
Is this a new TYPO3 standard? Last time i saw this it was still a timestamp without decimal signs.

sebastian-altenburg-nr replied:
It's still a timestamp, just uses underscores for better readability. According to the TYPO3 documentation, using timestamps as exception codes is acceptable. What’s your take on this? :)

CybotTM replied:
Yes, I’m aware they’re using timestamps, which is perfectly fine and that's what I wrote. However, incorporating underscores into the timestamp in this particular context seems unconventional. Typically, readability is the main reason for adding underscores to numeric values—but for an internal exception code, that advantage might not be significant.

Certainly, the current approach works and doesn’t pose any functional issues. However, for the sake of consistency with TYPO3’s coding guidelines—and to maintain a uniform style within our codebase—it might be worth avoiding different or unconventional formatting. **If you prefer to keep it as it is, that’s perfectly fine**. I simply wanted to mention that mixing numeric literal styles (e.g., with underscores) could potentially confuse developers who are used to standard TYPO3 conventions.

CybotTM commented on Classes/Domain/Model/ImageFileReference.php:
```diff
@@ -41,11 +41,11 @@ class ImageFileReference extends FileReference
      * Returns the title.
      *
-     * @return null|string
+     * @return string|null
```
@param and @return are superfluous when using type hints/declarations in the code.

sebastian-altenburg-nr replied:
That's good to know, thank you!

CybotTM replied:
Applies to the other source files as well ;-)

CybotTM reviewed (commented):
"NEXT-77: Prepare for rebase" and "NEXT-44: change composer.json" are no proper committ messages.

You are mixing required changes to make this  compatible to TYPO3 v13 with pure CS changes in single commits. This should be avoided.

Commit:
NEXT-77: Add AGPL license and update .gitignore

Added LICENSE.txt and added license to composer.json

Added phpstan-report.xml and .gitlab-ci.yml to .gitignore

Commit:
NEXT-77: Removing redundant annotations

CybotTM commented on Classes/Domain/Repository/ImageFileReferenceRepository.php:
```diff
@@ -35,29 +36,27 @@ class ImageFileReferenceRepository extends Repository
      * @var ConnectionPool
      */
-    private ConnectionPool $connectionPool;
+    private readonly ConnectionPool $connectionPool;
```
this is for PHP 8+ only, right?

then you should use Constructor property promotion

https://stitcher.io/blog/constructor-promotion-in-php-8

CybotTM replied:
oh, dunno if this works for child classes ...
